holoviews/core/util.py

```diff
@@ -1184,3 +1184,24 @@ def is_nan(x):
         return np.isnan(x)
     except:
         return False
+
+
+def bound_range(vals, density, TOL=10e-9):
```

Comment:
Will be glad to get rid of it. Code looks hairy and it uses the ``np.diff`` approach which isn't particularly efficient. Then there are the tolerance issues...